urllib3/response.py

```diff
@@ -319,6 +319,8 @@ def from_httplib(ResponseCls, r, **response_kw):
         with ``original_response=r``.
         """
         headers = r.msg
+        validate_headers(headers)
```

Comment:
Should this do anything with the response value?

Maybe `assert_headers` would be a better name, and have it raise an exception if it fails? Then we can catch/log in the handler with more context rather than in the validator itself.